modules/security/src/main/java/io/oasp/module/security/common/base/accesscontrol/AbstractAccessControlProvider.java

```diff
@@ -86,16 +86,15 @@ protected void checkForCyclicDependencies(AccessControlGroup group, Set<AccessCo
    */
   protected void collectAccessControls(AccessControlGroup group, Set<AccessControlGroup> toplevelGroups) {
 
-    // TODO hohwille Java HashMap buggy???
```

Comment:
I am very sure I out commented this code on purpose. Hence I expect this PR to introduce a bug.
Maybe the hashCode of the objects in the Set changed after they had been added...